### DISCUSSION TITLE
enum 的类型问题

### DISCUSSION THREAD

**Discussion by chenos** [alibaba/formily#2795] [Q&A 2.x] [answered]
```ts
type SchemaEnum<Message> = Array<
  | string
  | number
  | { label: Message; value: any; [key: string]: any }
  | { key: any; title: Message; [key: string]: any }
>
```

现在的 schema.enum 限制了只能是 {label,value} 或 {key,title} 这两种，能否再开放一点？

比如 Ant Deign 的 Select、TreeSelect、Cascader 等一些组件都支持 fieldNames 转换，enum 可能并不是 {label,value} 或 {key,title} 格式。

**Comment by janryWang** [comment, marked as answer]
不是有any的吗？对于其他扩展结构，也不会报类型错误呀

**Reply by chenos** [reply]
比如我传的是 { id, displayName } 就不行了

**Reply by janryWang** [reply]
不行是啥意思

**Reply by chenos** [reply]
https://codesandbox.io/s/distracted-maxwell-7s0le?file=/App.tsx

比如上面例子的 schema 会报错。

**Reply by janryWang** [reply]
没看到报错

**Reply by chenos** [reply]
因为加了 ts ignore，给忽略了，代码 54 行

**Reply by chenos** [reply]
加了 ISchema，看起来就明显了。

现在的 SchemaEnum，如果元素是 object 的情况，只能是 {label,value, ....} 或 {key,title, ....}。比如：

```ts
{ label: Message; value: any; [key: string]: any } // 这里虽然加了 any，但是 label 和 value 是必填的。
{ key: any; title: Message; [key: string]: any } // 这里 key 和 title 也必须是必填天的，any 只针对扩展的其他属性。
```

这个类型提示感觉并不十分合理，比如截图这样就报错了。

![image](https://user-images.githubusercontent.com/2993310/151119852-f203ef37-9b60-4a9b-adf6-3cf15356e9ff.png)

**Reply by janryWang** [reply]
嗯，已经改了，下个版本修复
